controller/resource.go

```diff
@@ -225,31 +226,15 @@ func (s *ServiceInfo) Reconcile() error {
 // Upsert creates a token for the task if one doesn't already exist
 // and updates the secret with the contents of the token.
 func (s *ServiceInfo) Upsert() error {
-	secretName := s.secretName()
-
-	// Get current secret from AWS.
-	currSecretValue, err := s.SecretsManagerClient.GetSecretValue(&secretsmanager.GetSecretValueInput{SecretId: aws.String(secretName)})
+	currSecret, err := s.upsertSecret()
 	if err != nil {
-		return fmt.Errorf("retrieving secret: %w", err)
-	}
-	var currSecret tokenSecretJSON
-	err = json.Unmarshal([]byte(*currSecretValue.SecretString), &currSecret)
-	if err != nil {
-		return fmt.Errorf("unmarshalling secret value: %w", err)
+		return fmt.Errorf("upserting secret: %w", err)
 	}
 
-	var currToken *api.ACLToken
-	// If we already have an accessor ID, we'll check if this token exists in Consul first.
-	// We don't care if the token value is empty or not in this case.
-	// If token value is empty, then it's an empty secret, and we should update it with the token.
-	// If token value is non-empty it indicates that something is corrupted, and we should update the token.
-	if currSecret.AccessorID != "" {
-		// Read the token with this Accessor ID from Consul.
-		currToken, _, err = s.ConsulClient.ACL().TokenRead(currSecret.AccessorID, nil)
-
-		if err != nil && !isACLNotFoundError(err) {
-			return fmt.Errorf("reading existing token: %w", err)
-		}
+	currToken, _, err := s.ConsulClient.ACL().TokenRead(currSecret.AccessorID, nil)
+
+	if err != nil && !isACLNotFoundError(err) {
+		return fmt.Errorf("reading existing token: %w", err)
```

Comment:
I'm leaving this around since there is a test for this code and it isn't causing any problems, but it really shouldn't be necessary. During this same reconcile cycle, a token is verified to not exist.